src/Utilities/TypeTraits.hpp

```diff
@@ -1429,3 +1431,29 @@ using function_info = tt_detail::function_info_impl<F>;
 template <typename T, size_t = 0>
 using identity_t = T;
 }  // namespace tt
+
+namespace cpp20 {
+/*!
+ * \ingroup Utilities
+ * Reimplementation of std::swap that is constexpr;
+ * taken from the LLVM source at
+ * https://github.com/llvm-mirror/libcxx/blob/master/include/type_traits
+ */
+template <class T>
+constexpr void swap(T& a, T& b) noexcept {
```

Comment:
Should be in Algorithm

Reply:
I’ll move it...only here because that’s where it is in the clang source, and I was trying to follow that.